superset-frontend/src/views/datasetList/Modal.tsx

```diff
@@ -0,0 +1,102 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import React from 'react';
+import styled from '@superset-ui/style';
+import { Modal as BaseModal } from 'react-bootstrap';
+import { t } from '@superset-ui/translation';
+import Button from './Button';
+
+type Callback = (...args: any[]) => void;
```

Comment:
This is probably reused in a bunch of different files, should we pull it out into a global types file?

Reply:
I'd say maybe make each callback more explicit instead... For example, here it could be 

```ts
interface ModalProps {
  ...
  onHide: BaseModal['onHide'] | Button['onClick'];
  onSave: Button['onClick'];
  ...
}
```

or if we don't actually want to pass any arguments to the callback, name it

```ts
type EmptyCallback = () => void;
```

and update component accordingly

```ts
export default function Modal({
  children,
  disableSave,
  onHide,
  onSave,
  show,
  title,
}: Props) {
  const onHideHandler = useCallback(() => if (onHide) onHide(), [onHide]);
  const onSaveHandler = useCallback(() => if (onSave) onSave(), [onSave]);
  return (
    <StyledModal show={show} onHide={onHideHandler} bsSize="lg">
      <StyledModalHeader closeButton>
        <BaseModal.Title>
          <Title>{title}</Title>
        </BaseModal.Title>
      </StyledModalHeader>
      <StyledModalBody>{children}</StyledModalBody>
      <StyledModalFooter>
        <span className="float-right">
          <Button title={t('Cancel')} bsStyle="secondary" onClick={onHideHandler} />
          <Button
            bsStyle="primary"
            disabled={disableSave}
            onClick={onSaveHandler}
            title={t('Add')}
          />
        </span>
      </StyledModalFooter>
    </StyledModal>
  );
}
```